cdc/redo/manager.go

```diff
@@ -98,302 +160,171 @@ func (s *statefulRts) setFlushed(flushed model.Ts) {
 	atomic.StoreUint64(&s.flushed, flushed)
 }
 
-func (s *statefulRts) checkAndSetUnflushed(unflushed model.Ts) {
+func (s *statefulRts) checkAndSetUnflushed(unflushed model.Ts) (changed bool) {
 	for {
 		old := atomic.LoadUint64(&s.unflushed)
 		if old > unflushed {
-			return
+			return false
 		}
 		if atomic.CompareAndSwapUint64(&s.unflushed, old, unflushed) {
 			break
 		}
 	}
+	return true
 }
 
-// ManagerImpl manages redo log writer, buffers un-persistent redo logs, calculates
-// redo log resolved ts. It implements LogManager interface.
-type ManagerImpl struct {
-	changeFeedID model.ChangeFeedID
-	enabled      bool
+// logManager manages redo log writer, buffers un-persistent redo logs, calculates
+// redo log resolved ts. It implements DDLManager and DMLManager interface.
+type logManager struct {
+	enabled bool
+	cfg     *writer.LogWriterConfig
+	writer  writer.RedoLogWriter
 
-	opts *ManagerOptions
+	rwlock sync.RWMutex
```

Comment:
Seems rwlock only protects the writer now. So we can make `writer` thread-safe to remove `rwlock`. I guess it makes it more clean.

Reply:
This is for compatibility with older file writer, and it also ensures that chann could be reliably shut down.